Dockerfile

```diff
@@ -102,6 +103,7 @@ RUN rpmdev-setuptree \
    && echo -e -n "${GRUB_SET_PRIVATE_VAR:+%bcond_without grub_set_private_var\n}" >> .bconds \
    && echo -e -n "${SYSTEMD_NETWORKD:+%bcond_without systemd_networkd\n}" >> .bconds \
    && echo -e -n "${UNIFIED_CGROUP_HIERARCHY:+%bcond_without unified_cgroup_hierarchy\n}" >> .bconds \
+   && echo -e -n "${XFS_DATA_PARTITION:+%bcond_without xfs_data_partition\n}" >> .bconds \
```

Comment:
I added this for if we need to do conditional specfile things. I didn't end up adding `xfsprogs` conditionally to the image based upon this but we could if we want. If we don't see the need for this I can remove it as well.

Reply:
Should it be something more generic? Like `DATA_PARTITION_FORMAT=<the-format>`? Wouldn't OOTB benefit from this? Say I'm a weird developer that wants to use btrfs for my data partition in my OOT variant.

Reply:
> I didn't end up adding xfsprogs conditionally to the image based upon this but we could if we want.

It's fine to add the bcond support here even if we aren't using it, just for consistency. I did the same in #3097.